apps/backend/src/modules/ai-assistant/ai-assistant.service.ts

```diff
@@ -0,0 +1,163 @@
+import { type OpenAi, OpenAIRoleKey } from "~/libs/modules/open-ai/open-ai.js";
+import { type UserDto } from "~/libs/types/types.js";
+import { type CategoryService } from "~/modules/categories/categories.js";
+import { type OnboardingRepository } from "~/modules/onboarding/onboarding.js";
+import { type TaskService } from "~/modules/tasks/tasks.js";
+
+import {
+	generateChangeTaskSuggestionsResponse,
+	generateExplainTaskSuggestionsResponse,
+	generateQuestionsAnswersPrompt,
+	generateScoresResponse,
+	generateTaskSuggestionsResponse,
+	runChangeTaskByCategoryOptions,
+	runExplainTaskOptions,
+	runInitialThreadOptions,
+	runTaskByCategoryOptions,
+} from "./libs/helpers/helpers.js";
+import {
+	type AIAssistantRequestDto,
+	type AIAssistantResponseDto,
+	type SelectedCategory,
+	type TaskCreateDto,
+	type TaskDto,
+	type ThreadMessageCreateDto,
+} from "./libs/types/types.js";
+
+type Constructor = {
+	categoryService: CategoryService;
+	onboardingRepository: OnboardingRepository;
+	openAi: OpenAi;
+	taskService: TaskService;
+};
+
+class AIAssistantService {
+	private categoryService: CategoryService;
+	private onboardingRepository: OnboardingRepository;
+	private openAi: OpenAi;
+	private taskService: TaskService;
+
+	public constructor({
+		categoryService,
+		onboardingRepository,
+		openAi,
+		taskService,
+	}: Constructor) {
+		this.openAi = openAi;
```

Comment:
```suggestion
		this.openAI = openAI;
```
Make sure to fix everywhere

Reply:
`const generateScoresResponse = (
	aiResponse: OpenAIResponseMessage,
): AIAssistantResponseDto | null => {`

Only this left, but I'm not sure do I need to rename aiResponse to AIResponse